apps/etterlatte-statistikk/src/main/kotlin/statistikk/database/AktivitetspliktRepo.kt

```diff
@@ -0,0 +1,169 @@
+        yearMonth: YearMonth,
+    ): StatistikkAktivitet? =
+        datasource.connection.use { connection ->
+            val statement =
+                connection.prepareStatement(
+                    """
+                    SELECT sak_id, registrert, avdoed_doedsmaaned, unntak, brukers_aktivitet, 
+                        aktivitetsgrad, varig_unntak
+                    FROM aktivitetsplikt
+                    WHERE sak_id = ?
+                    AND registrert_maaned <= ?
+                    """.trimIndent(),
+                )
+            statement.setLong(1, sakId)
+            statement.setString(2, yearMonth.toString())
+            statement.executeQuery().singleOrNull {
+                somStatistikkAkvititet()
+            }
+        }
+
+    private fun ResultSet.somStatistikkAkvititet(): StatistikkAktivitet =
+        StatistikkAktivitet(
+            sakId = getLong("sak_id"),
+            registrert = getTidspunkt("registrert"),
+            avdoedDoedsmaaned = getString("avdoed_doedsmaaned").let { YearMonth.parse(it) },
+            unntak = getString("unntak").let { objectMapper.readValue(it) },
+            brukersAktivitet = getString("brukers_aktivitet").let { objectMapper.readValue(it) },
+            aktitetsgrad = getString("aktivitetsgrad").let { objectMapper.readValue(it) },
+            harVarigUnntak = getBoolean("varig_unntak"),
+        )
+}
+
+data class StatistikkAktivitet(
+    val sakId: Long,
+    val registrert: Tidspunkt,
+    val avdoedDoedsmaaned: YearMonth,
+    val unntak: List<PeriodisertAktivitetspliktopplysning>,
+    val brukersAktivitet: List<PeriodisertAktivitetspliktopplysning>,
+    val aktitetsgrad: List<PeriodisertAktivitetspliktopplysning>,
+    val harVarigUnntak: Boolean,
+) {
+    companion object {
+        fun fra(dto: AktivitetspliktDto): StatistikkAktivitet =
+            StatistikkAktivitet(
+                sakId = dto.sakId,
+                registrert = Tidspunkt.now(),
+                avdoedDoedsmaaned = dto.avdoedDoedsmaaned,
+                unntak = dto.unntak.map { PeriodisertAktivitetspliktopplysning.fra(it) },
+                brukersAktivitet = dto.brukersAktivitet.map { PeriodisertAktivitetspliktopplysning.fra(it) },
+                aktitetsgrad = dto.aktivitetsgrad.map { PeriodisertAktivitetspliktopplysning.fra(it) },
+                harVarigUnntak = dto.unntak.any { it.unntak == UnntakFraAktivitetsplikt.FOEDT_1963_ELLER_TIDLIGERE_OG_LAV_INNTEKT },
+            )
+    }
+}
+
+/**
+ * Mapper enumverdier til konstante strenger. Gjør at en eventuell endring / migrering av enums ikke
+ * påvirker mappingen til statistikk.
+ */
+data class PeriodisertAktivitetspliktopplysning(
+    val opplysning: String,
+    val fom: LocalDate?,
+    val tom: LocalDate?,
+) {
+    companion object {
+        fun fra(opplysning: UnntakFraAktivitetDto): PeriodisertAktivitetspliktopplysning =
+            PeriodisertAktivitetspliktopplysning(
+                opplysning =
```

Comment:
Er `opplysning = opplysning.unntak.name` like enkelt? Så ingen forskjell i teksten. Gjelder også mappingen under.

Reply:
Det er mer for å unngå at vi endrer statistikkverdiene utilsiktet, og må ha et bevisst forhold til de - men ja den er kanskje i overkant defensiv